## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#6ed3fa1f-fcad-4a19-8731-33158a20ed80) |
| mint-large-bucket.sh | :x: [more...](#c300761c-2497-4880-bd36-1dbc5e9cbc69) |
| mint-compression-dist-xl.sh | :x: [more...](#c8612137-3906-40fc-99d7-367e63a9bf18) |
| mint-dist-xl.sh | :x: [more...](#0d6a66d7-008d-40bc-8e6b-fedf87f1ce33) |
<a name="6ed3fa1f-fcad-4a19-8731-33158a20ed80"></a>
### 7453-5438625/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.54:32541
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 55b56a0a446e:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-u41j9rcqaspits",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 330,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.54:32541/aws-sdk-go-test-u41j9rcqaspits: dial tcp 72.28.97.54:32541: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="0d6a66d7-008d-40bc-8e6b-fedf87f1ce33"></a>
### 7453-5438625/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.60:31153
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp cca8d5542479:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-xvzbzutsml0zdz",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 350,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.60:31153/aws-sdk-go-test-xvzbzutsml0zdz: dial tcp 72.28.97.60:31153: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="c300761c-2497-4880-bd36-1dbc5e9cbc69"></a>
### 7453-5438625/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:32545
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 31bcadb3197c:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 4 minutes and 2 seconds
(2/14) Running aws-sdk-java tests ... FAILED in 51 seconds
com.amazonaws.services.s3.model.AmazonS3Exception: Server not initialized, please try again. (Service: Amazon S3; Status Code: 503; Error Code: XMinioServerNotInitialized; Request ID: 15917709A87ECA16; S3 Extended Request ID: 15917709A87ECA16), S3 Extended Request ID: 15917709A87ECA16
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1630)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1302)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1056)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:743)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:717)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:699)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:667)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:649)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:513)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4330)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4277)
	at com.amazonaws.services.s3.AmazonS3Client.createBucket(AmazonS3Client.java:1025)
	at io.minio.awssdk.tests.FunctionalTests.initTests(Unknown Source)
	at io.minio.awssdk.tests.FunctionalTests.main(Unknown Source)

Executed 1 out of 14 tests successfully.
```
<a name="c8612137-3906-40fc-99d7-367e63a9bf18"></a>
### 7453-5438625/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:32576
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 79bb3c973f4a:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 46 seconds
(4/14) Running aws-sdk-ruby tests ... done in 4 seconds
(5/14) Running awscli tests ... done in 51 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 29 seconds
(8/14) Running minio-dotnet tests ... done in 41 seconds
(9/14) Running minio-go tests ... done in 53 seconds
(10/14) Running minio-java tests ... done in 2 minutes and 3 seconds
(11/14) Running minio-js tests ... done in 39 seconds
(12/14) Running minio-py tests ... done in 3 minutes and 47 seconds
(13/14) Running s3cmd tests ... FAILED in 1 seconds
{
  "name": "s3cmd",
  "duration": "447",
  "function": "test_put_object",
  "status": "FAIL",
  "error": "\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  following lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n\nInvoked as: /usr/local/bin/s3cmd --config /tmp/.s3cmd-6391/s3cfg rm s3://s3cmd-test-bucket-4472/s3cmd-test-object-13144\nProblem: <class 'OSError: [Errno 113] No route to host\nS3cmd:   2.0.2\npython:   3.5.2 (default, Nov 12 2018, 13:43:14) \n[GCC 5.4.0 20160609]\nenvironment LANG=C.UTF-8\n\nTraceback (most recent call last):\n  File \"/usr/local/bin/s3cmd\", line 3092, in <module>\n    rc = main()\n  File \"/usr/local/bin/s3cmd\", line 3001, in main\n    rc = cmd_func(args)\n  File \"/usr/local/bin/s3cmd\", line 639, in cmd_object_del\n    rc = subcmd_object_del_uri(uri_str)\n  File \"/usr/local/bin/s3cmd\", line 766, in subcmd_object_del_uri\n    response = s3.object_delete(S3Uri(item['object_uri_str']))\n  File \"/usr/local/lib/python3.5/dist-packages/S3/S3.py\", line 759, in object_delete\n    response = self.send_request(request)\n  File \"/usr/local/lib/python3.5/dist-packages/S3/S3.py\", line 1258, in send_request\n    conn = ConnMan.get(self.get_hostname(resource['bucket']))\n  File \"/usr/local/lib/python3.5/dist-packages/S3/ConnMan.py\", line 253, in get\n    conn.c.connect()\n  File \"/usr/lib/python3.5/http/client.py\", line 849, in connect\n    (self.host,self.port), self.timeout, self.source_address)\n  File \"/usr/lib/python3.5/socket.py\", line 711, in create_connection\n    raise err\n  File \"/usr/lib/python3.5/socket.py\", line 702, in create_connection\n    sock.connect(sa)\nOSError: [Errno 113] No route to host\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  above lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
}

Executed 12 out of 14 tests successfully.
```

--- a/cmd/fs-v1.go
+++ b/cmd/fs-v1.go
@@ -250,7 +250,9 @@ func (fs *FSObjects) StorageInfo(ctx context.Context) StorageInfo {
 		used = atomic.LoadUint64(&fs.totalUsed)
 	}
 	storageInfo := StorageInfo{
-		Used: used,
+		Used:      used,
+		Total:     di.Total,
+		Available: di.Free,
 	}
 	storageInfo.Backend.Type = BackendFS
 	return storageInfo

--- a/cmd/metrics.go
+++ b/cmd/metrics.go
@@ -142,6 +142,26 @@ func (c *minioCollector) Collect(ch chan<- prometheus.Metric) {
 		float64(s.Used),
 	)
 
+	// Total disk available space seen by Minio server instance
+	ch <- prometheus.MustNewConstMetric(
+		prometheus.NewDesc(
+			prometheus.BuildFQName("minio", "disk", "storage_available_bytes"),
+			"Total disk available space seen by Minio server instance",
+			nil, nil),
+		prometheus.GaugeValue,
+		float64(s.Available),
+	)
+
+	// Total disk space seen by Minio server instance
+	ch <- prometheus.MustNewConstMetric(
+		prometheus.NewDesc(
+			prometheus.BuildFQName("minio", "disk", "storage_total_bytes"),
+			"Total disk space seen by Minio server instance",
+			nil, nil),
+		prometheus.GaugeValue,
+		float64(s.Total),
+	)
+
 	// Minio Total Disk/Offline Disk
 	ch <- prometheus.MustNewConstMetric(
 		prometheus.NewDesc(

--- a/cmd/object-api-datatypes.go
+++ b/cmd/object-api-datatypes.go
@@ -41,6 +41,10 @@ const (
 type StorageInfo struct {
 	Used uint64 // Used total used per tenant.
 
+	Total uint64 // Total disk space.
+
+	Available uint64 // Total disk space available.
+
 	// Backend type.
 	Backend struct {
 		// Represents various backend types, currently on FS and Erasure.

--- a/cmd/xl-sets.go
+++ b/cmd/xl-sets.go
@@ -307,6 +307,8 @@ func (s *xlSets) StorageInfo(ctx context.Context) StorageInfo {
 	for _, set := range s.sets {
 		lstorageInfo := set.StorageInfo(ctx)
 		storageInfo.Used = storageInfo.Used + lstorageInfo.Used
+		storageInfo.Total = storageInfo.Total + lstorageInfo.Total
+		storageInfo.Available = storageInfo.Available + lstorageInfo.Available
 		storageInfo.Backend.OnlineDisks = storageInfo.Backend.OnlineDisks + lstorageInfo.Backend.OnlineDisks
 		storageInfo.Backend.OfflineDisks = storageInfo.Backend.OfflineDisks + lstorageInfo.Backend.OfflineDisks
 	}

--- a/cmd/xl-v1.go
+++ b/cmd/xl-v1.go
@@ -120,18 +120,23 @@ func getStorageInfo(disks []StorageAPI) StorageInfo {
 		return StorageInfo{}
 	}
 
-	// Combine all disks to get total usage.
-	var used uint64
+	// Combine all disks to get total usage
+	var used, total, available uint64
 	for _, di := range validDisksInfo {
 		used = used + di.Used
+		total = total + di.Total
+		available = available + di.Free
 	}
 
 	_, sscParity := getRedundancyCount(standardStorageClass, len(disks))
 	_, rrscparity := getRedundancyCount(reducedRedundancyStorageClass, len(disks))
 
 	storageInfo := StorageInfo{
-		Used: used,
+		Used:      used,
+		Total:     total,
+		Available: available,
 	}
+
 	storageInfo.Backend.Type = BackendErasure
 	storageInfo.Backend.OnlineDisks = onlineDisks
 	storageInfo.Backend.OfflineDisks = offlineDisks

--- a/pkg/madmin/API.md
+++ b/pkg/madmin/API.md
@@ -130,64 +130,65 @@ Sends a service action command to service - possible actions are restarting and 
 ### ServerInfo() ([]ServerInfo, error)
 Fetches information for all cluster nodes, such as server properties, storage information, network statistics, etc.
 
-| Param | Type | Description |
-|---|---|---|
-|`si.Addr` | _string_ | Address of the server the following information is retrieved from. |
-|`si.ConnStats` | _ServerConnStats_ | Connection statistics from the given server. |
-|`si.HTTPStats` | _ServerHTTPStats_ | HTTP connection statistics from the given server. |
-|`si.Properties` | _ServerProperties_ | Server properties such as region, notification targets. |
-|`si.Data.StorageInfo.Total`  | _int64_  | Total disk space. |
-|`si.Data.StorageInfo.Free`  | _int64_  | Free disk space. |
-|`si.Data.StorageInfo.Backend`| _struct{}_ | Represents backend type embedded structure. |
+| Param                           | Type               | Description                                                        |
+|---------------------------------|--------------------|--------------------------------------------------------------------|
+| `si.Addr`                       | _string_           | Address of the server the following information is retrieved from. |
+| `si.ConnStats`                  | _ServerConnStats_  | Connection statistics from the given server.                       |
+| `si.HTTPStats`                  | _ServerHTTPStats_  | HTTP connection statistics from the given server.                  |
+| `si.Properties`                 | _ServerProperties_ | Server properties such as region, notification targets.            |
+| `si.Data.StorageInfo.Used`      | _int64_            | Used disk space.                                                   |
+| `si.Data.StorageInfo.Total`     | _int64_            | Total disk space.                                                  |
+| `si.Data.StorageInfo.Available` | _int64_            | Available disk space.                                              |
+| `si.Data.StorageInfo.Backend`   | _struct{}_         | Represents backend type embedded structure.                        |
 
-| Param | Type | Description |
-|---|---|---|
-|`ServerProperties.Uptime`| _time.Duration_ | Total duration in seconds since server is running. |
-|`ServerProperties.Version`| _string_ | Current server version. |
-|`ServerProperties.CommitID` | _string_ | Current server commitID. |
-|`ServerProperties.Region` | _string_ | Configured server region. |
-|`ServerProperties.SQSARN` | _[]string_ | List of notification target ARNs. |
+| Param                       | Type            | Description                                        |
+|-----------------------------|-----------------|----------------------------------------------------|
+| `ServerProperties.Uptime`   | _time.Duration_ | Total duration in seconds since server is running. |
+| `ServerProperties.Version`  | _string_        | Current server version.                            |
+| `ServerProperties.CommitID` | _string_        | Current server commitID.                           |
+| `ServerProperties.Region`   | _string_        | Configured server region.                          |
+| `ServerProperties.SQSARN`   | _[]string_      | List of notification target ARNs.                  |
 
-| Param | Type | Description |
-|---|---|---|
-|`ServerConnStats.TotalInputBytes` | _uint64_ | Total bytes received by the server. |
-|`ServerConnStats.TotalOutputBytes` | _uint64_ | Total bytes sent by the server. |
+| Param                              | Type     | Description                         |
+|------------------------------------|----------|-------------------------------------|
+| `ServerConnStats.TotalInputBytes`  | _uint64_ | Total bytes received by the server. |
+| `ServerConnStats.TotalOutputBytes` | _uint64_ | Total bytes sent by the server.     |
 
-| Param | Type | Description |
-|---|---|---|
-|`ServerHTTPStats.TotalHEADStats`| _ServerHTTPMethodStats_ | Total statistics regarding HEAD operations |
-|`ServerHTTPStats.SuccessHEADStats`| _ServerHTTPMethodStats_ | Total statistics regarding successful HEAD operations |
-|`ServerHTTPStats.TotalGETStats`| _ServerHTTPMethodStats_ |  Total statistics regarding GET operations |
-|`ServerHTTPStats.SuccessGETStats`| _ServerHTTPMethodStats_ | Total statistics regarding successful GET operations |
-|`ServerHTTPStats.TotalPUTStats`| _ServerHTTPMethodStats_ | Total statistics regarding PUT operations |
-|`ServerHTTPStats.SuccessPUTStats`| _ServerHTTPMethodStats_ | Total statistics regarding successful PUT operations |
-|`ServerHTTPStats.TotalPOSTStats`| _ServerHTTPMethodStats_ | Total statistics regarding POST operations |
-|`ServerHTTPStats.SuccessPOSTStats`| _ServerHTTPMethodStats_ | Total statistics regarding successful POST operations |
-|`ServerHTTPStats.TotalDELETEStats`| _ServerHTTPMethodStats_ | Total statistics regarding DELETE operations |
-|`ServerHTTPStats.SuccessDELETEStats`| _ServerHTTPMethodStats_ | Total statistics regarding successful DELETE operations |
+| Param                                | Type                    | Description                                             |
+|--------------------------------------|-------------------------|---------------------------------------------------------|
+| `ServerHTTPStats.TotalHEADStats`     | _ServerHTTPMethodStats_ | Total statistics regarding HEAD operations              |
+| `ServerHTTPStats.SuccessHEADStats`   | _ServerHTTPMethodStats_ | Total statistics regarding successful HEAD operations   |
+| `ServerHTTPStats.TotalGETStats`      | _ServerHTTPMethodStats_ | Total statistics regarding GET operations               |
+| `ServerHTTPStats.SuccessGETStats`    | _ServerHTTPMethodStats_ | Total statistics regarding successful GET operations    |
+| `ServerHTTPStats.TotalPUTStats`      | _ServerHTTPMethodStats_ | Total statistics regarding PUT operations               |
+| `ServerHTTPStats.SuccessPUTStats`    | _ServerHTTPMethodStats_ | Total statistics regarding successful PUT operations    |
+| `ServerHTTPStats.TotalPOSTStats`     | _ServerHTTPMethodStats_ | Total statistics regarding POST operations              |
+| `ServerHTTPStats.SuccessPOSTStats`   | _ServerHTTPMethodStats_ | Total statistics regarding successful POST operations   |
+| `ServerHTTPStats.TotalDELETEStats`   | _ServerHTTPMethodStats_ | Total statistics regarding DELETE operations            |
+| `ServerHTTPStats.SuccessDELETEStats` | _ServerHTTPMethodStats_ | Total statistics regarding successful DELETE operations |
 
 
-| Param | Type | Description |
-|---|---|---|
-|`ServerHTTPMethodStats.Count` | _uint64_ | Total number of operations. |
-|`ServerHTTPMethodStats.AvgDuration` | _string_ | Average duration of Count number of operations. |
+| Param                               | Type     | Description                                     |
+|-------------------------------------|----------|-------------------------------------------------|
+| `ServerHTTPMethodStats.Count`       | _uint64_ | Total number of operations.                     |
+| `ServerHTTPMethodStats.AvgDuration` | _string_ | Average duration of Count number of operations. |
 
-| Param | Type | Description |
-|---|---|---|
-|`Backend.Type` | _BackendType_ | Type of backend used by the server currently only FS or Erasure. |
-|`Backend.OnlineDisks`| _int_ | Total number of disks online (only applies to Erasure backend), is empty for FS. |
-|`Backend.OfflineDisks` | _int_ | Total number of disks offline (only applies to Erasure backend), is empty for FS. |
-|`Backend.StandardSCData` | _int_ | Data disks set for standard storage class, is empty for FS. |
-|`Backend.StandardSCParity` | _int_ | Parity disks set for standard storage class, is empty for FS. |
-|`Backend.RRSCData` | _int_ | Data disks set for reduced redundancy storage class, is empty for FS. |
-|`Backend.RRSCParity` | _int_ | Parity disks set for reduced redundancy storage class, is empty for FS. |
-|`Backend.Sets` | _[][]DriveInfo_ | Represents topology of drives in erasure coded sets. |
+| Param                      | Type            | Description                                                                       |
+|----------------------------|-----------------|-----------------------------------------------------------------------------------|
+| `Backend.Type`             | _BackendType_   | Type of backend used by the server currently only FS or Erasure.                  |
+| `Backend.OnlineDisks`      | _int_           | Total number of disks online (only applies to Erasure backend), is empty for FS.  |
+| `Backend.OfflineDisks`     | _int_           | Total number of disks offline (only applies to Erasure backend), is empty for FS. |
+| `Backend.StandardSCData`   | _int_           | Data disks set for standard storage class, is empty for FS.                       |
+| `Backend.StandardSCParity` | _int_           | Parity disks set for standard storage class, is empty for FS.                     |
+| `Backend.RRSCData`         | _int_           | Data disks set for reduced redundancy storage class, is empty for FS.             |
+| `Backend.RRSCParity`       | _int_           | Parity disks set for reduced redundancy storage class, is empty for FS.           |
+| `Backend.Sets`             | _[][]DriveInfo_ | Represents topology of drives in erasure coded sets.                              |
 
-| Param | Type | Description |
-|---|---|---|
-|`DriveInfo.UUID`| _string_ | Unique ID for each disk provisioned by server format. |
-|`DriveInfo.Endpoint` | _string_ | Endpoint location of the remote/local disk. |
-|`DriveInfo.State` | _string_ | Current state of the disk at endpoint. |
+| Param                | Type     | Description                                           |
+|----------------------|----------|-------------------------------------------------------|
+| `DriveInfo.UUID`     | _string_ | Unique ID for each disk provisioned by server format. |
+| `DriveInfo.Endpoint` | _string_ | Endpoint location of the remote/local disk.           |
+| `DriveInfo.State`    | _string_ | Current state of the disk at endpoint.                |
 
  __Example__
 
@@ -468,7 +469,7 @@ Lists all users on Minio server.
 __Example__
 
 ``` go
-	users, err := madmClnt.ListUsers(); 
+	users, err := madmClnt.ListUsers();
     if err != nil {
 		log.Fatalln(err)
 	}

--- a/pkg/madmin/info-commands.go
+++ b/pkg/madmin/info-commands.go
@@ -51,6 +51,10 @@ type DriveInfo HealDriveInfo
 type StorageInfo struct {
 	Used uint64 // Total used spaced per tenant.
 
+	Available uint64 // Total available space.
+
+	Total uint64 // Total disk space.
+
 	// Backend type.
 	Backend struct {
 		// Represents various backend types, currently on FS and Erasure.

